src/Cache/Server/Server.php

```diff
@@ -0,0 +1,576 @@
+			: get_transient( self::UPDATING_KEYS_TRANSIENT );
+
+		is_array( $keys ) or $keys = [];
+
+		$keys[] = $key;
+
+		return $is_network
+			? set_site_transient( self::UPDATING_KEYS_TRANSIENT, $keys )
+			: set_transient( self::UPDATING_KEYS_TRANSIENT, $keys );
+	}
+
+	/**
+	 * Remove the given key from transient storage to mark given key again available for updates.
+	 *
+	 * @param string $key
+	 * @param bool   $is_network
+	 *
+	 * @return bool
+	 */
+	private function mark_not_updating( string $key, bool $is_network ): bool {
+
+		$keys = $is_network
+			? get_site_transient( self::UPDATING_KEYS_TRANSIENT )
+			: get_transient( self::UPDATING_KEYS_TRANSIENT );
+
+		return $is_network
+			? set_site_transient( self::UPDATING_KEYS_TRANSIENT, array_diff( (array) $keys, [ $key ] ) )
+			: set_transient( self::UPDATING_KEYS_TRANSIENT, array_diff( (array) $keys, [ $key ] ) );
+	}
+
+	/**
+	 * Use transients to check if the given key is currently being updated in a update request, to prevent
+	 * multiple concurrent updates.
+	 *
+	 * @param string $key
+	 * @param bool   $is_network
+	 *
+	 * @return bool
+	 */
+	private function is_updating( $key, bool $is_network ): bool {
+
+		$keys = $is_network
+			? get_site_transient( self::UPDATING_KEYS_TRANSIENT )
+			: get_transient( self::UPDATING_KEYS_TRANSIENT );
+
+		return $keys && in_array( $key, (array) $keys, true );
+	}
+
+	/**
+	 * Queue given key to be updated in a HTTP request.
+	 * The first time it is called adds an action on shutdown that will actually process the queue
+	 * and send updating HTTP requests.
+	 *
+	 * @param string $key
+	 * @param int    $time_to_live
+	 * @param bool   $is_network
+	 */
+	private function queue_update( string $key, int $time_to_live, bool $is_network ) {
+
+		if ( $this->in_spawn_queue ) {
+			return;
+		}
+
+		if ( ! $this->spawn_queue && ! did_action( 'shutdown' ) ) {
+			add_action( 'shutdown', function () {
+
+				$this->in_spawn_queue = true;
+				$this->spawn_queue();
+				$this->spawn_queue = [];
+			}, 50 );
+		}
+
+		$this->spawn_queue[ $key ] = [ $key, $time_to_live, $is_network ];
+	}
+
+	/**
+	 * Send multiple HTTP request to refresh registered cache items.
+	 */
+	private function spawn_queue(): array {
+
+		if ( ! $this->spawn_queue || ! $this->in_spawn_queue ) {
+			return [];
+		}
+
+		$requests     = [];
+		$network_keys = $keys = [];
+
+		foreach ( $this->spawn_queue as list( $key, $time_to_live, $is_network ) ) {
+
+			if ( $this->is_spawning( $key, $is_network ) ) {
+				continue;
+			}
+
+			$is_network ? $network_keys[] = $key : $keys[] = $key;
+
+			$requests[ $key ] = [
+				'url'     => home_url(),
+				'headers' => [
+					self::HEADER_KEY => $key,
+					self::HEADER_TTL => $time_to_live,
+				],
+				'data'    => '',
+				'cookies' => [],
+			];
+		}
+
+		$this->mark_spawning( true, ...$network_keys );
+		$this->mark_spawning( false, ...$keys );
+
+		\Requests::request_multiple(
```

Comment:
Only break the array elements, but the let the rest inline: `\Requests::request_multiple( $requests, [`.